content/developer/execution_model.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Workflow Execution Model
+description:
+weight: 70
+---
+
+## Running an execution
+
+An execution is started by a user, by sending a POST request to the REST API's /executions endpoint.
+
+1. The REST API creates an execution object in the database, including a random execution token.
+1. The restservice sends "start-workflow" message to the management queue on rabbitmq. That message contains at least the following fields:
+
+        {
+            'type': 'workflow',
+            'task_name': import path of the workflow function (eg. "cloudify.plugins.workflows.install"),
+            'task_id': execution id,
+            'workflow_id': id of the workflow (eg. "install"),
+            'blueprint_id': blueprint id,
+            'deployment_id': deployment id,
+            'runtime_only_evaluation': boolean,
+            'execution_id': execution id,
+            'bypass_maintenance': boolean,
+            'dry_run': boolean,
+            'is_system_workflow': boolean,
+            'wait_after_fail': boolean,
+            'resume': boolean,
+            'execution_token': execution token,
+            'plugin': details of the plugin containing the workflow function (dict)
+        }
+
+1. The management worker receives the "start-workflow" message from rabbitmq, and runs a "dispatcher" subprocess to handle it.
+1. The subprocess loads and executes the workflow function. Most workflow functions create a tasks graph and execute it.
+1. The subprocess periodically checks execution state to react to the state changing to CANCELLED.
+1. The workflow function might store the tasks graph and the operations, to allow resuming.
+1. For every operation in the tasks graph, the dispatcher process sends "start-operation" messages:
+    - the operation is transitioned to the SENDING state, and then SENT
+    - the message is sent to the exchange `<agent_name>` for host_agent tasks, or `cloudify.management` for central_deployment_agent tasks
+    - the dispatcher process starts listening for the task result on the `<agent_name>_response_<task_id>`.
```

Comment:
Is this the same (with agent_name = cloudify.management) for central_deployment_agent?

Reply:
reworded the previous line